File: code/scala/pipelines/load_acc_status_silver/code/job/graph/load.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from job.config.ConfigStore import *
from job.udfs.UDFs import *

def load(spark: SparkSession) -> DataFrame:

    if Config.fabricName == "anshuman2":
        return spark.read\
            .schema(
              StructType([
                StructField("acc_id", StringType(), True), StructField("person_id", StringType(), True), StructField("product_id", StringType(), True), StructField("business_date", StringType(), True), StructField("balance", StringType(), True)
            ])
            )\
            .option("header", True)\
            .option("sep", ",")\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/acc_status/bronze/")
    else:
        raise Exception("No valid dataset present to read fabric")

    if Config.fabricName == "anshuman":
        return spark.read\
            .schema(
              StructType([
                StructField("acc_id", StringType(), True), StructField("person_id", StringType(), True), StructField("product_id", StringType(), True), StructField("business_date", StringType(), True), StructField("balance", StringType(), True)
            ])
            )\
            .option("header", True)\
            .option("sep", ",")\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/acc_status/bronze/")
    else:
        raise Exception("No valid dataset present to read fabric")

File: code/scala/pipelines/load_transactions/code/src/main/scala/graph/load.scala
package graph

import io.prophecy.libs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import config.ConfigStore._

object load {

  def apply(spark: SparkSession): DataFrame = {
    Config.fabricName match {
      case "anshuman2" =>
        spark.read
          .format("parquet")
          .option("mergeSchema", true)
          .schema(
            StructType(
              Array(
                StructField("acc_id",        IntegerType,   false),
                StructField("business_date", DateType,      false),
                StructField("tran_amount",   DoubleType,    false),
                StructField("tran_type",     StringType,    false),
                StructField("tran_ts",       TimestampType, false)
              )
            )
          )
          .load(
            "dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/transactions/bronze/"
          )
      case _ =>
        throw new Exception("No valid dataset present to read fabric")
    }
  }

}
